subsys/net/ip/Kconfig.tcp

```diff
@@ -231,7 +231,7 @@ config NET_TCP_ISN_RFC6528
 	depends on NET_TCP
 	select MBEDTLS
 	select MBEDTLS_MD
-	select MBEDTLS_MAC_MD5_ENABLED
+	select MBEDTLS_MAC_SHA256_ENABLED
```

Comment:
We now claim to support `CONFIG_NET_TCP_ISN_RFC6528` but we are not because we do not use MD5 as said by the RFC. IMHO if we want to use SHA256, we should invent a new kconfig option for it. If we cannot use the new RFC for this, then lets have some other named option for it.

Reply:
> We now claim to support CONFIG_NET_TCP_ISN_RFC6528 but we are not because we do not use MD5 as said by the RFC

I'm not sure we are not compliant with RFC6528 with this change. [Section 3](https://datatracker.ietf.org/doc/html/rfc6528#section-3) states that:

> Since MD5 is faster than other "stronger" alternatives, and is used in virtually all existing implementations of this algorithm, we consider that use of MD5 in the specified algorithm is acceptable. However, implementations should consider the trade-offs involved in using functions with stronger security properties, and employ them if it is deemed appropriate.

To me this does not impose the use of MD5, but rather suggests it as an option and it does not prevent the possibility to use SHA256 instead. But that might be only my understanding of that paragraph and this might differ from your. Wdyt?

Reply:
Indeed, good point. In this case no need to re-invent the wheel, we can just change the code to use the sha256.